src/textual/widgets/_key_panel.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import annotations
+
+from collections import defaultdict
+from typing import TYPE_CHECKING
+
+from rich.table import Table
+from rich.text import Text
+
+from ..app import ComposeResult
+from ..binding import Binding
+from ..containers import VerticalScroll
+from ..reactive import reactive
+from ..widgets import Static
+
+if TYPE_CHECKING:
+    from ..screen import Screen
+
+
+class BindingsTable(Static):
+    """A widget to display bindings."""
+
+    COMPONENT_CLASSES = {"bindings-table--key", "bindings-table--description"}
+
+    DEFAULT_CSS = """
+    BindingsTable {
+        width: auto;
+        height: auto;
+    }
+    """
+
+    upper_case_keys = reactive(False)
+    """Upper case key display."""
+    ctrl_to_caret = reactive(True)
+    """Convert 'ctrl+' prefix to '^'."""
```

Comment:
Just thinking out loud for potential future updates: I'd love if this was:

```python
ctrl_repr = reactive("^")
and_repr = reactive("")
```

Then I could choose the styling easily. e.g. if I wanted emacs style bindings:

```python
foo.ctrl_repr = 'c'
foo.and_repr = '-'
```

Which would render `ctrl+x` as `c-x`.

Reply:
However, I think this would be defined at the app level - so we don't need to define this for both our Footer and for the Bindings table.